stl/src/locale0.cpp

```diff
@@ -135,8 +135,39 @@ __PURE_APPDOMAIN_GLOBAL locale::id ctype<unsigned short>::id(0);
 
 __PURE_APPDOMAIN_GLOBAL locale::id codecvt<unsigned short, char, mbstate_t>::id(0);
 
+#define _Compiler_barrier() _STL_DISABLE_DEPRECATED_WARNING _ReadWriteBarrier() _STL_RESTORE_DEPRECATED_WARNING
+
+#if defined(_M_ARM) || defined(_M_ARM64) || defined(_M_ARM64EC)
+#define _Memory_barrier()             __dmb(0xB) // inner shared data memory barrier
+#define _Compiler_or_memory_barrier() _Memory_barrier()
+#elif defined(_M_IX86) || defined(_M_X64)
+// x86/x64 hardware only emits memory barriers inside _Interlocked intrinsics
+#define _Compiler_or_memory_barrier() _Compiler_barrier()
+#else // ^^^ x86/x64 / unsupported hardware vvv
+#error Unsupported hardware
+#endif // hardware
+
 _MRTIMP2_PURE const locale& __CLRCALL_PURE_OR_CDECL locale::classic() { // get reference to "C" locale
-    _Init();
+#if !defined(_M_CEE_PURE)
+    const auto mem = reinterpret_cast<const intptr_t*>(&locale::_Locimp::_Clocptr);
+    intptr_t as_bytes;
+#ifdef _WIN64
+#ifdef _M_ARM
+    as_bytes = __ldrexd(_Mem);
+#else // ^^^ ARM64 / x64 vvv
+    as_bytes = __iso_volatile_load64(mem);
+#endif // ^^^ x64

```

Comment:
Comment nitpick: Should end with `^^^` too.

Reply:
This `#ifdef`/`#else`/`#endif` is removed in light of the previous comment